nilearn/_utils/data_gen.py

```diff
@@ -333,47 +426,153 @@ def generate_fake_fmri(shape=(10, 11, 12), length=17, kind="noise",
+                                       rk=3,
+                                       affine=np.eye(4),
+                                       random_state=0):
+    """
+    Generate random fmri data and design matrices of given shapes.
+
+    Parameters
+    ----------
+    shapes : list of 4D tuples
+        Shapes of the fmri data to be generated.
+
+    rk : int, optional
+        Number of columns in the design matrix to be generated.
+        Default=3.
+
+    affine : numpy.ndarray, optional
+        Affine of returned images.
+        Default=np.eye(4).
+
+    random_state : int or numpy.random.RandomState instance, optional
+        Random number generator, or seed.
+        Default=0.
+
+    Returns
+    -------
+    mask : nibabel.Nifti1Image
+        3D, Mask giving non-zero voxels.
+
+    fmri_data : List
+        list of nibabel.Nifti1Image
+
+    design_matrices : List
+        list of pd.DataFrame
+
+    """
     fmri_data = []
     design_matrices = []
+    rand_gen = check_random_state(random_state)
     for i, shape in enumerate(shapes):
-        data = np.random.randn(*shape)
+        data = rand_gen.randn(*shape)
         data[1:-1, 1:-1, 1:-1] += 100
         fmri_data.append(Nifti1Image(data, affine))
-        columns = np.random.choice(list(string.ascii_lowercase),
-                                   size=rk,
-                                   replace=False)
-        design_matrices.append(pd.DataFrame(np.random.randn(shape[3], rk),
-                                            columns=columns))
-    mask = Nifti1Image((np.random.rand(*shape[:3]) > .5).astype(np.int8),
+        columns = rand_gen.choice(list(string.ascii_lowercase),
+                                  size=rk,
+                                  replace=False)
+        design_matrices.append(
+            pd.DataFrame(rand_gen.randn(shape[3], rk), columns=columns))
+    mask = Nifti1Image((rand_gen.rand(*shape[:3]) > .5).astype(np.int8),
                        affine)
     return mask, fmri_data, design_matrices
 
 
-def write_fake_fmri_data_and_design(shapes, rk=3, affine=np.eye(4)):
+def write_fake_fmri_data_and_design(shapes,
+                                    rk=3,
+                                    affine=np.eye(4),
+                                    random_state=0):
+    """
+    Generate random fmri data and design matrices and write them to disk.
+
+    Parameters
+    ----------
+    shapes : List
+        List of shapes in tuple format
+
+    rk : int, optional
+        Number of columns in the design matrix to be generated.
+        Default=3.
+
+    affine : numpy.ndarray, optional
+        Affine of returned images.
+        Default=np.eye(4).
+
+    random_state : int or numpy.random.RandomState instance, optional
+        Random number generator, or seed.
+        Default=0.
+
+    Returns
+    -------
+    mask_file : str
+        path to mask file.
+
+    fmri_files : List
+        list of str.
```

Comment:
This would be more appropriate as the type, and the description could be something like "list of paths to the generated fmri files" or something like that. WDYT?

Reply:
Yes I agree that's better, will change it